tcl/response.tcl

```diff
@@ -1,18 +1,30 @@
 namespace eval qc::response {
 
-    namespace export status record message action extend
+    namespace export status record message action extend data
     namespace ensemble create -unknown {
         response_subcommand_map
     }
 
+    proc data {tson} {
+        #| Sets the JSON response object data property using tson
+        global data
+        dict set data data $tson
+    }
+
     proc extend {name args} {
         #| Extends the JSON response with an object named $name with properties defined in $args.
         if { [llength $args] % 2 != 0 } {
             return -code error "Usage: qc::response extend name key value ?key value ...?"
         }
         global data
-        dict for {key value} $args {
-            dict set data $name $key $value
+        if { $name eq "data" } {
```

Comment:
This code change is required in order to keep the existing behaviour, in case any code currently uses `qc::response extend data name value` (which is the existing usage).

I would expect this usage to be deprecated in favour of `qc::response extend ..." going forward.